### PR TITLE
Add gRPC support

### PR DESCRIPTION
**Pull request by Dean-Coakley**
Add gRPC support.

- [x] gRPC support
- [x] Updated docker files
- [x] Example
- [x] Documentation
- [x] Logging

### CODE REVIEW

File: examples/grpc-services/README.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Grpc support
```
Comment (pleshakov) [inline examples/grpc-services/README.md]: Grpc -> gRPC

File: nginx-controller/Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM nginx:1.13.8
+FROM nginx:1.13.10
```
Comment (pleshakov) [inline nginx-controller/Dockerfile]: please 1.13.12

File: nginx-controller/DockerfileForAlpine [hunk 3]
```diff
@@ -1,4 +1,4 @@
-FROM nginx:1.13.8-alpine
+FROM nginx:1.13.10-alpine
```
Comment (pleshakov) [inline nginx-controller/DockerfileForAlpine]: 1.13.12

File: nginx-controller/nginx/configurator.go [hunk 4]
```diff
@@ -517,6 +534,21 @@ func getSSLServices(ingEx *IngressEx) map[string]bool {
...
 }
 
+func getGrpcServices(ingCfg *Config, ingEx *IngressEx) map[string]bool {
+	grpcServices := make(map[string]bool)
+
+	if services, exists := ingEx.Ingress.Annotations["nginx.org/grpc-services"]; exists {
+		for _, svc := range strings.Split(services, ",") {
+			grpcServices[svc] = true
+		}
+		if len(grpcServices) > 0 && !ingCfg.HTTP2 {
```
Comment (pleshakov) [inline nginx-controller/nginx/configurator.go]: please move the condition and the log to `generateNginxCfg`

File: nginx-controller/nginx/templates/nginx-plus.ingress.tmpl [hunk 5]
```diff
@@ -12,9 +12,11 @@ upstream {{$upstream.Name}} {
 
 {{range $server := .Servers}}
 server {
+	{{if not $server.GRPCOnly}}
```
Comment (isaachawley) [inline nginx-controller/nginx/templates/nginx-plus.ingress.tmpl]: whitespace?

File: nginx-controller/nginx/templates/nginx-plus.ingress.tmpl [hunk 6]
```diff
@@ -81,20 +111,24 @@ server {
...
+
 		client_max_body_size {{$location.ClientMaxBodySize}};
 		proxy_set_header Host $host;
 		proxy_set_header X-Real-IP $remote_addr;
 		proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
 		proxy_set_header X-Forwarded-Host $host;
 		proxy_set_header X-Forwarded-Port $server_port;
+
 		proxy_set_header X-Forwarded-Proto {{if $server.RedirectToHTTPS}}https{{else}}$scheme{{end}};
+
```
Comment (isaachawley) [inline nginx-controller/nginx/templates/nginx-plus.ingress.tmpl]: whitespace?

File: nginx-controller/nginx/templates/nginx.ingress.tmpl [hunk 7]
```diff
@@ -89,10 +122,14 @@ server {
...
 		{{- end}}
+
 		{{if $location.SSL}}
 		proxy_pass https://{{$location.Upstream.Name}}{{$location.Rewrite}};
 		{{else}}
 		proxy_pass http://{{$location.Upstream.Name}}{{$location.Rewrite}};
 		{{end}}
+
+		{{end}}
+
```
Comment (isaachawley) [inline nginx-controller/nginx/templates/nginx.ingress.tmpl]: whitespace?

**Pull request merged**